src/virtual-machine.js

```diff
@@ -1033,6 +1034,25 @@ class VirtualMachine extends EventEmitter {
         return null;
     }
 
+    /**
+     * Reorder target by index. Return whether a change was made.
+     * @param {!string} targetIndex Index of the target.
+     * @param {!number} newIndex index that the target should be moved to.
+     * @returns {boolean} Whether a target was reordered.
+     */
+    reorderTarget (targetIndex, newIndex) {
```

Comment:
I'm wondering if it would be more useful to have this function take a targetId instead of a targetIndex? I guess then it would be called 'setTargetOrder' or 'setTargetIndex' or something.

Reply:
I initially had this, but the way the drag/drop code is working with respect to the GUI it is easiest  to just treat it similar to costume/sounds, just referring by index. It would just take more work to look up the targets id first.